astropy/io/fits/hdu/compressed.py

```diff
@@ -694,6 +718,31 @@ def match_header(cls, header):
             # Compression is supported but disabled; just pass silently (#92)
             return False
 
+    def _strip_default_extname(self, image_header):
+        """Remove default COMPRESSED_IMAGE value for EXTNAME.
+
+        Parameters
+        ----------
+        image_header : `~astropy.io.fits.Header`
+            Header to be modified.
+
+        Notes
+        -----
+        If there are multiple EXTNAME keys, only the one containing the
+        default value is stripped.
+        """
+        if ('EXTNAME' in image_header and
+                image_header['EXTNAME'] == 'COMPRESSED_IMAGE'):
+            # In some rare cases there might be a COMPRESSED_IMAGE in
+            # the header and non-default EXTNAME. Only delete the default.
+            indices = image_header._keyword_indices['EXTNAME']
+            to_remove = []
+            for index in indices:
+                if image_header[index] == 'COMPRESSED_IMAGE':
+                    to_remove.append(index)
+            for index in to_remove:
+                del image_header[index]
```

Comment:
This doesn't work if COMPRESSED_IMAGE is present more than once. Because when you remove the first occurrence it modifies the indices of indices, so the indices that you have in `to_remove` are no more valid.
The solution is probably to delete just the first COMPRESSED_IMAGE found, if there are more than one, well ...

Reply:
I meant to say:
```python
for index in reversed(to_remove):
   del image_header[index]
```
since presumably that does work without causing any reindexing problems.

Reply:
if somehow we have two COMPRESSED_IMAGE entries and then the real one we'd still want to delete both and if we didn't it would report COMPRESSED_IMAGE as now. I don't think what I've got is going to cause any particular slowdown because in almost every case it would only loop once and even if we had two COMPRESSED_IMAGE and a real one it would only loop three times.

Reply:
In fact the test file I have included here has 4 EXTNAME headers in it (two of them are COMPRESSED_IMAGE).

Reply:
4 EXTNAME :scream: 
I guess the indices in `_keyword_indices` should be ordered, at least that's the case when you read a file, though you could still manually insert a duplicate card before another one. Maybe a `reversed(sorted(to_remove))` to be sure ?